core/templates/dev/head/pages/Base.ts

```diff
@@ -55,6 +58,11 @@ oppia.controller('Base', [
       $scope.currentLang = response.language;
     });
 
+    // This is done outside the service to prevent
+    // circular dependency issues.
+    $http.get('/csrfhandler').then(function(response) {
+      CsrfTokenService.setToken(response.data.token);
+    });
```

Comment:
Since we use jQuery now can we also call the service here?

Reply:
I think we this should be removed altogether since the call will be made by App.ts

Reply:
On second thoughts, I think it would be faster to call getToken so it has more time to resolve

Reply:
I think you don't need to call `getToken` from here if it is not needed because `getToken` is called from App.ts which is loaded quite early anyway.

Reply:
The `getToken` request is only called before making a `PUT` or `POST` request, then the token get's cached. Meaning that the first `POST` or `PUT` request on a page will take time to get completed because it will be making two requests.

Reply:
Also, that caused some e2e test failure because the time taken to send both requests might become large